adapters/factMapper/builder_test.go

```diff
@@ -26,17 +26,20 @@ func TestBuilderInvariants(t *testing.T) {
 	testutil.TestBuilderInvariants(b, t)
 }
 
-func TestNoRules(t *testing.T) {
+func setup(t *testing.T, bc BuilderConfig, ac AdapterConfig) adapters.FactConverter {
 	b := NewBuilder()
-	b.Configure(b.DefaultBuilderConfig())
+	b.Configure(&bc)
 
-	rules := make(map[string]string)
-	var aa adapters.Adapter
-	var err error
-	if aa, err = b.NewAdapter(&AdapterConfig{Rules: rules}); err != nil {
+	a, err := b.NewAdapter(&ac)
+	if err != nil {
 		t.Error("Expected to successfully create a mapper")
 	}
-	a := aa.(adapters.FactConverter)
+	return a.(adapters.FactConverter)
+}
+
+func TestNoRules(t *testing.T) {
+	rules := make(map[string]string)
+	a := setup(t, BuilderConfig{}, AdapterConfig{Rules: rules})
```

Comment:
Perhaps for another PR, but there is a lot of repetition in this TestNoRules() code.  Can we just have a cases or tests struct with a rules map and loop through with the same check and error message?